tensorflow/python/autograph/tests/loop_scoping_test.py

```diff
@@ -38,6 +38,23 @@ def while_with_local_var(x):
     x -= 1
   return s
 
+def for_with_lambda_iter(l):
+  fns = []
+  results = []
+  for i in l:
+    fns.append(lambda: i)
```

Comment:
There are many uses where `i` here would be a non-tensor, so when added to the tf.while_loop's variables it would break existing code. I think we need to think of a way to roll this out in a gradual and compatible fashion, for example by writing something like

```
  for i in l:
    fns.append(lambda: i)  # autograph: nonlocal-lambda
```
or perhaps the converse

```
  for i in l:
    fns.append(lambda: i)  # autograph: local-lambda
```

The exact method for annotating it can vary. It might be simpler to write something like `fns.append(tf.autograph.nonlocal(lambda: i))`.